--- net-tools/Makefile~	2015-10-02 10:24:26.000000000 +0200
+++ net-tools/Makefile	2015-10-02 10:26:37.964020572 +0200
@@ -132,11 +132,14 @@ distcheck:	dist
 		@printf "\nThe tarball is ready to go:\n%s\n" "`du -b net-tools-$(RELEASE).tar.xz`"
 
 config.h: 	config.in Makefile
-		@echo "Configuring the Linux net-tools (NET-3 Base Utilities)..." ; echo
-		@if [ config.status -nt config.in ]; \
-			then ./configure.sh config.status; \
-		   else ./configure.sh config.in; \
-		 fi
+		@if [ ! -f config.h ]; \
+			then echo "Configuring the Linux net-tools (NET-3 Base Utilities)..." ; \
+			echo ; \
+			if [ config.status -nt config.in ]; \
+				then ./configure.sh config.status; \
+			else ./configure.sh config.in; \
+			fi ; \
+		fi
 
 
 version.h:	Makefile
