diff -urNp -x '*.orig' net-tools-2.10.org/man/en_US/netstat.8 net-tools-2.10/man/en_US/netstat.8
--- net-tools-2.10.org/man/en_US/netstat.8	2021-01-07 00:22:35.000000000 +0100
+++ net-tools-2.10/man/en_US/netstat.8	2023-12-02 23:35:08.827986462 +0100
@@ -47,9 +47,9 @@ netstat \- Print network connections, ro
 .RB [ \-\-continuous | \-c ]
 .P
 .B netstat
-.RB { \-\-interfaces | \-i }
+.RB { \-\-interfaces | \-I | \-i }
 .RB [ \-\-all | \-a ]
-.RB [ \-\-extend | \-e  [ \-\-extend | \-e] ]
+.RB [ \-\-extend | \-e  ]
 .RB [ \-\-verbose | \-v ]
 .RB [ \-\-program | \-p ]
 .RB [ \-\-numeric | \-n ]
@@ -128,8 +128,8 @@ and
 produce the same output.
 .SS "\-\-groups, \-g"
 Display multicast group membership information for IPv4 and IPv6.
-.SS "\-\-interfaces, \-i"
-Display a table of all network interfaces.
+.SS "\-\-interfaces=\fIiface \fR, \fB\-I=\fIiface \fR, \fB\-i"
+Display a table of all network interfaces, or the specified \fIiface\fR.
 .SS "\-\-masquerade, \-M"
 Display a list of masqueraded connections.
 .SS "\-\-statistics, \-s"
diff -urNp -x '*.orig' net-tools-2.10.org/netstat.c net-tools-2.10/netstat.c
--- net-tools-2.10.org/netstat.c	2023-12-02 23:35:08.762985691 +0100
+++ net-tools-2.10/netstat.c	2023-12-02 23:35:08.827986462 +0100
@@ -149,6 +149,7 @@ static char *Release = RELEASE, *Signatu
 #define E_IOCTL -3
 
 int flag_int = 0;
+char *flag_int_name = NULL;
 int flag_rou = 0;
 int flag_mas = 0;
 int flag_sta = 0;
@@ -2132,6 +2133,7 @@ static int rfcomm_info(void)
 
 static int iface_info(void)
 {
+    struct interface *ife = NULL;
     if (skfd < 0) {
 	if ((skfd = sockets_open(0)) < 0) {
 	    perror("socket");
@@ -2144,7 +2146,11 @@ static int iface_info(void)
 	printf(_("Iface             MTU    RX-OK RX-ERR RX-DRP RX-OVR    TX-OK TX-ERR TX-DRP TX-OVR Flg\n"));
     }
 
-    if (for_all_interfaces(do_if_print, &flag_all) < 0) {
+    if (flag_int_name) {
+        ife = lookup_interface(flag_int_name);
+        do_if_print(ife, &flag_all);
+    }
+    else if (for_all_interfaces(do_if_print, &flag_all) < 0) {
 	perror(_("missing interface information"));
 	exit(1);
     }
@@ -2171,10 +2177,11 @@ static void usage(int rc)
     FILE *fp = rc ? stderr : stdout;
     fprintf(fp, _("usage: netstat [-vWeenNcCF] [<Af>] -r         netstat {-V|--version|-h|--help}\n"));
     fprintf(fp, _("       netstat [-vWnNcaeol] [<Socket> ...]\n"));
-    fprintf(fp, _("       netstat { [-vWeenNac] -i | [-cnNe] -M | -s [-6tuw] }\n\n"));
+    fprintf(fp, _("       netstat { [-vWeenNac] -I[<Iface>] | -i | [-cnNe] -M | -s [-6tuw] }\n\n"));
 
     fprintf(fp, _("        -r, --route              display routing table\n"));
     fprintf(fp, _("        -i, --interfaces         display interface table\n"));
+    fprintf(fp, _("        -I, --interfaces=<Iface> display interface table for <Iface>\n"));
     fprintf(fp, _("        -g, --groups             display multicast group memberships\n"));
     fprintf(fp, _("        -s, --statistics         display networking statistics (like SNMP)\n"));
 #if HAVE_FW_MASQUERADE
@@ -2217,7 +2224,7 @@ int main
     {
 	AFTRANS_OPTS,
 	{"version", 0, 0, 'V'},
-	{"interfaces", 0, 0, 'i'},
+	{"interfaces", 2, 0, 'I'},
 	{"help", 0, 0, 'h'},
 	{"route", 0, 0, 'r'},
 #if HAVE_FW_MASQUERADE
@@ -2261,7 +2268,7 @@ int main
     getroute_init();		/* Set up AF routing support */
 
     afname[0] = '\0';
-    while ((i = getopt_long(argc, argv, "A:CFMacdeghilnNoprsStuUvVWw2fx64?Z", longopts, &lop)) != EOF)
+    while ((i = getopt_long(argc, argv, "A:CFMacdeghiI::lnNoprsStuUvVWw2fx64?Z", longopts, &lop)) != EOF)
 	switch (i) {
 	case -1:
 	    break;
@@ -2302,6 +2309,13 @@ int main
 	case 'p':
 	    flag_prg++;
 	    break;
+	case 'I':
+	    if (optarg && strcmp(optarg, "(null)"))
+		if (optarg[0] == '=') optarg++;
+	    if (optarg && strcmp(optarg, "(null)"))
+		flag_int_name = strdup(optarg);
+	    flag_int++;
+	    break;
 	case 'i':
 	    flag_int++;
 	    break;
