--- net-tools-1.60/rarp.c~	2001-04-08 19:05:05.000000000 +0200
+++ net-tools-1.60/rarp.c	2013-02-16 22:41:25.888096075 +0100
@@ -225,7 +225,7 @@
 	case 'h':
 	    usage();
 	case 'V':
-	    fprintf(stderr, version_string);
+	    fprintf(stderr, "%s", version_string);
 	    exit(E_VERSION);
 	    break;
 	case 'v':
--- net-tools-1.60/slattach.c~	2000-10-28 12:59:41.000000000 +0200
+++ net-tools-1.60/slattach.c	2013-02-16 22:42:41.059884440 +0100
@@ -565,7 +565,7 @@
 	  "[-c cmd] [-s speed] [-p protocol] tty | -\n"
 	  "       slattach -V | --version\n";
 
-  fprintf(stderr, usage_msg);
+  fprintf(stderr, "%s", usage_msg);
   exit(1);
 }
 
--- net-tools-1.60/mii-tool.c~	2013-02-16 22:42:58.786976322 +0100
+++ net-tools-1.60/mii-tool.c	2013-02-16 22:43:17.657430178 +0100
@@ -271,7 +271,7 @@
 
     if (opt_watch) {
 	if (opt_log) {
-	    syslog(LOG_INFO, buf);
+	    syslog(LOG_INFO, "%s", buf);
 	} else {
 	    char s[20];
 	    time_t t = time(NULL);
