--- net-tools-1.57/ifconfig.c.bug9215	Sat Oct  7 11:05:58 2000
+++ net-tools-1.57/ifconfig.c	Sat Oct  7 11:06:59 2000
@@ -306,14 +306,16 @@
     /* The next argument is either an address family name, or an option. */
     if ((ap = get_aftype(*spp)) == NULL)
 	ap = get_aftype(DFLT_AF);
-    else {
+    else
+	spp++;
+
+    if (ap) {
 	/* XXX: should print the current setup if no args left, but only 
 	   for this family */
-	spp++;
 	addr_family = ap->af;
     }
 
-    if (sockets_open(addr_family) < 0) {
+    if ((skfd = sockets_open(addr_family)) < 0) {
 	perror("family socket");
 	exit(1);
     }
