--- net-tools-1.57/lib/net-support.h.bug9129	Sat May 20 14:27:25 2000
+++ net-tools-1.57/lib/net-support.h	Sat Oct  7 11:19:50 2000
@@ -92,14 +92,17 @@
 #define RTACTION_FLUSH 4
 #define RTACTION_SHOW  5
 
-#define FLAG_EXT      3		/* AND-Mask */
-#define FLAG_NUM      4
-#define FLAG_SYM      8
-#define FLAG_CACHE   16
-#define FLAG_FIB     32
-#define FLAG_VERBOSE 64
+#define FLAG_EXT       3		/* AND-Mask */
+#define FLAG_NUM_HOST  4
+#define FLAG_NUM_PORT  8
+#define FLAG_NUM_USER 16
+#define FLAG_NUM     (FLAG_NUM_HOST|FLAG_NUM_PORT|FLAG_NUM_USER)
+#define FLAG_SYM      32
+#define FLAG_CACHE    64
+#define FLAG_FIB     128
+#define FLAG_VERBOSE 256
 
-extern int ip_masq_info(int numeric, int ext);
+extern int ip_masq_info(int numeric_host, int numeric_port, int ext);
 
 extern int INET_rprint(int options);
 extern int INET6_rprint(int options);
--- net-tools-1.57/lib/inet6_gr.c.bug9129	Sat Jan  9 10:54:50 1999
+++ net-tools-1.57/lib/inet6_gr.c	Sat Oct  7 11:19:50 2000
@@ -259,7 +259,7 @@
 int INET6_rprint(int options)
 {
     int ext = options & FLAG_EXT;
-    int numeric = options & (FLAG_NUM | FLAG_SYM);
+    int numeric = options & (FLAG_NUM_HOST | FLAG_SYM);
     int rc = E_INTERN;
 
     if (options & FLAG_FIB)
--- net-tools-1.57/lib/inet_gr.c.bug9129	Sun Jan  2 17:53:44 2000
+++ net-tools-1.57/lib/inet_gr.c	Sat Oct  7 11:19:50 2000
@@ -444,7 +444,7 @@
 int INET_rprint(int options)
 {
     int ext = options & FLAG_EXT;
-    int numeric = options & (FLAG_NUM | FLAG_SYM);
+    int numeric = options & (FLAG_NUM_HOST | FLAG_SYM);
     int rc = E_INTERN;
 
     if (options & FLAG_FIB)
--- net-tools-1.57/lib/ipx_gr.c.bug9129	Sat Jan  9 10:55:11 1999
+++ net-tools-1.57/lib/ipx_gr.c	Sat Oct  7 11:19:50 2000
@@ -33,7 +33,7 @@
 int IPX_rprint(int options)
 {
     /* int ext = options & FLAG_EXT; */
-    int numeric = options & FLAG_NUM;
+    int numeric = options & FLAG_NUM_HOST;
     char buff[1024];
     char net[128], router_net[128];
     char router_node[128];
--- net-tools-1.57/lib/netrom_gr.c.bug9129	Sat Jan  9 10:55:20 1999
+++ net-tools-1.57/lib/netrom_gr.c	Sat Oct  7 11:19:50 2000
@@ -37,7 +37,7 @@
     char buffer[256];
     int qual, n, w;
     /*int ext = options & FLAG_EXT;
-       int numeric = options & FLAG_NUM; */
+       int numeric = options & FLAG_NUM_HOST; */
 
     f1 = fopen(_PATH_PROCNET_NR_NODES, "r");
     if (!f1) perror(_PATH_PROCNET_NR_NODES);
--- net-tools-1.57/lib/masq_info.c.bug9129	Sat Jun 12 19:04:49 1999
+++ net-tools-1.57/lib/masq_info.c	Sat Oct  7 11:19:50 2000
@@ -57,7 +57,8 @@
 static struct aftype *ap;	/* current address family       */
 static int has_pdelta;
 
-static void print_masq(struct masq *ms, int numeric, int ext)
+static void print_masq(struct masq *ms, int numeric_host, int numeric_port,
+		       int ext)
 {
     unsigned long minutes, seconds, sec100s;
 
@@ -77,12 +78,12 @@
 	    printf("%10lu %5hd     - ", ms->initseq,
 		   ms->delta);
     }
-    printf("%-20s ", ap->sprint((struct sockaddr *) &(ms->src), numeric));
-    printf("%-20s ", ap->sprint((struct sockaddr *) &(ms->dst), numeric));
+    printf("%-20s ", ap->sprint((struct sockaddr *) &(ms->src), numeric_host));
+    printf("%-20s ", ap->sprint((struct sockaddr *) &(ms->dst), numeric_host));
 
-    printf("%s -> ", get_sname(ms->sport, ms->proto, numeric));
-    printf("%s", get_sname(ms->dport, ms->proto, numeric));
-    printf(" (%s)\n", get_sname(ms->mport, ms->proto, numeric));
+    printf("%s -> ", get_sname(ms->sport, ms->proto, numeric_port));
+    printf("%s", get_sname(ms->dport, ms->proto, numeric_port));
+    printf(" (%s)\n", get_sname(ms->mport, ms->proto, numeric_port));
 }
 
 
@@ -144,7 +145,7 @@
 }
 
 
-int ip_masq_info(int numeric, int ext)
+int ip_masq_info(int numeric_host, int numeric_port, int ext)
 {
     FILE *f;
     int i;
@@ -204,7 +205,7 @@
 	    break;
 	}
 	for (i = 0; i < ntotal; i++)
-	    print_masq(&(mslist[i]), numeric, ext);
+	    print_masq(&(mslist[i]), numeric_host, numeric_port, ext);
 	if (mslist)
 	    free(mslist);
 
--- net-tools-1.57/man/en_US/netstat.8.bug9129	Sun May 21 16:16:38 2000
+++ net-tools-1.57/man/en_US/netstat.8	Sat Oct  7 11:54:53 2000
@@ -23,6 +23,7 @@
 .RB [ \-\-listening | \-l ]
 .RB [ \-\-all | \-a ]
 .RB [ \-\-numeric | \-n ]
+.RB [ \-\-numeric-hosts ] [ \-\-numeric-ports ] [ \-\-numeric-ports ]
 .RB [ \-\-symbolic | \-N ]
 .RB [ \-\-extend | \-e  [ \-\-extend | \-e] ]
 .RB [ \-\-timers | \-o ]
@@ -36,6 +37,7 @@
 .RB [ \-\-extend | \-e  [ \-\-extend | \-e] ]
 .RB [ \-\-verbose | \-v ]
 .RB [ \-\-numeric | \-n ]
+.RB [ \-\-numeric-hosts ] [ \-\-numeric-ports ] [ \-\-numeric-ports ]
 .RB [ \-\-continuous | \-c]
 .P
 .B netstat
@@ -46,17 +48,20 @@
 .RB [ \-\-verbose | \-v ]
 .RB [ \-\-program | \-p ]
 .RB [ \-\-numeric | \-n ]
+.RB [ \-\-numeric-hosts ] [ \-\-numeric-ports ] [ \-\-numeric-ports ]
 .RB [ \-\-continuous | \-c]
 .P
 .B netstat
 .RB { \-\-groups | \-g }
 .RB [ \-\-numeric | \-n ]
+.RB [ \-\-numeric-hosts ] [ \-\-numeric-ports ] [ \-\-numeric-ports ]
 .RB [ \-\-continuous | \-c]
 .P
 .B netstat
 .RB { \-\-masquerade | \-M }
 .RB [ \-\-extend | \-e ]
 .RB [ \-\-numeric | \-n ]
+.RB [ \-\-numeric-hosts ] [ \-\-numeric-ports ] [ \-\-numeric-ports ]
 .RB [ \-\-continuous | \-c]
 .P
 .B netstat
@@ -110,6 +115,16 @@
 .SS "\-\-numeric , \-n"
 Show numerical addresses instead of trying to determine symbolic host, port
 or user names.
+.SS "\-\-numeric-hosts"
+shows numerical host addresses but does not affect the resolution of
+port or user names.
+.SS "\-\-numeric-ports"
+shows numerical port numbers but does not affect the resolution of
+host or user names.
+.SS "\-\-numeric-users"
+shows numerical user IDs but does not affect the resolution of host or
+port names.
+
 .SS "\-\-protocol=\fIfamily \fR, \fB\-A"
 Specifies the address families (perhaps better described as low level
 protocols) for which connections are to be shown.
--- net-tools-1.57/netstat.c.bug9129	Sun May 28 11:19:15 2000
+++ net-tools-1.57/netstat.c	Sat Oct  7 11:19:50 2000
@@ -525,7 +525,7 @@
     struct passwd *pw;
 
     if (flag_exp > 1) {
-	if (!flag_not && ((pw = getpwuid(uid)) != NULL))
+	if (!(flag_not & FLAG_NUM_USER) && ((pw = getpwuid(uid)) != NULL))
 	    printf("%-10s ", pw->pw_name);
 	else
 	    printf("%-10d ", uid);
@@ -763,14 +763,17 @@
     safe_strncpy(rem_addr, ap->sprint((struct sockaddr *) &remaddr, flag_not),
 		 sizeof(rem_addr));
     if (flag_all || (flag_lst && !rem_port) || (!flag_lst && rem_port)) {
-	snprintf(buffer, sizeof(buffer), "%s", get_sname(htons(local_port), "tcp", flag_not));
+	snprintf(buffer, sizeof(buffer), "%s",
+		 get_sname(htons(local_port), "tcp",
+			   flag_not & FLAG_NUM_PORT));
 
 	if ((strlen(local_addr) + strlen(buffer)) > 22)
 	    local_addr[22 - strlen(buffer)] = '\0';
 
 	strcat(local_addr, ":");
 	strcat(local_addr, buffer);
-	snprintf(buffer, sizeof(buffer), "%s", get_sname(htons(rem_port), "tcp", flag_not));
+	snprintf(buffer, sizeof(buffer), "%s",
+		 get_sname(htons(rem_port), "tcp", flag_not & FLAG_NUM_PORT));
 
 	if ((strlen(rem_addr) + strlen(buffer)) > 22)
 	    rem_addr[22 - strlen(buffer)] = '\0';
@@ -911,13 +914,16 @@
     {
         safe_strncpy(local_addr, ap->sprint((struct sockaddr *) &localaddr, 
 					    flag_not), sizeof(local_addr));
-	snprintf(buffer, sizeof(buffer), "%s", get_sname(htons(local_port), "udp", flag_not));
+	snprintf(buffer, sizeof(buffer), "%s",
+		 get_sname(htons(local_port), "udp",
+			   flag_not & FLAG_NUM_PORT));
 	if ((strlen(local_addr) + strlen(buffer)) > 22)
 	    local_addr[22 - strlen(buffer)] = '\0';
 	strcat(local_addr, ":");
 	strcat(local_addr, buffer);
 
-	snprintf(buffer, sizeof(buffer), "%s", get_sname(htons(rem_port), "udp", flag_not));
+	snprintf(buffer, sizeof(buffer), "%s",
+		 get_sname(htons(rem_port), "udp", flag_not & FLAG_NUM_PORT));
         safe_strncpy(rem_addr, ap->sprint((struct sockaddr *) &remaddr, 
 					  flag_not), sizeof(rem_addr));
 	if ((strlen(rem_addr) + strlen(buffer)) > 22)
@@ -1025,7 +1031,9 @@
 
     if (flag_all || (notnull(remaddr) && !flag_lst) || (!notnull(remaddr) && flag_lst))
     {
-	snprintf(buffer, sizeof(buffer), "%s", get_sname(htons(local_port), "raw", flag_not));
+	snprintf(buffer, sizeof(buffer), "%s",
+		 get_sname(htons(local_port), "raw",
+			   flag_not & FLAG_NUM_PORT));
         safe_strncpy(local_addr, ap->sprint((struct sockaddr *) &localaddr, 
 					    flag_not), sizeof(local_addr));
 	if ((strlen(local_addr) + strlen(buffer)) > 22)
@@ -1033,7 +1041,8 @@
 	strcat(local_addr, ":");
 	strcat(local_addr, buffer);
 
-	snprintf(buffer, sizeof(buffer), "%s", get_sname(htons(rem_port), "raw", flag_not));
+	snprintf(buffer, sizeof(buffer), "%s",
+		 get_sname(htons(rem_port), "raw", flag_not & FLAG_NUM_PORT));
         safe_strncpy(rem_addr, ap->sprint((struct sockaddr *) &remaddr, 
 					  flag_not), sizeof(rem_addr));
 	if ((strlen(rem_addr) + strlen(buffer)) > 22)
@@ -1412,7 +1421,7 @@
 
 	printf("IPX   %6ld %6ld %-26s %-26s %-5s", txq, rxq, sad, dad, st);
 	if (flag_exp > 1) {
-	    if (!flag_not && ((pw = getpwuid(uid)) != NULL))
+	    if (!(flag_not & FLAG_NUM_USER) && ((pw = getpwuid(uid)) != NULL))
 		printf(" %-10s", pw->pw_name);
 	    else
 		printf(" %-10d", uid);
@@ -1475,6 +1484,9 @@
 #endif
     fprintf(stderr, _("        -v, --verbose            be verbose\n"));
     fprintf(stderr, _("        -n, --numeric            dont resolve names\n"));
+    fprintf(stderr, _("        --numeric-hosts          dont resolve host names\n"));
+    fprintf(stderr, _("        --numeric-ports          dont resolve port names\n"));
+    fprintf(stderr, _("        --numeric-users          dont resolve user names\n"));
     fprintf(stderr, _("        -N, --symbolic           resolve hardware names\n"));
     fprintf(stderr, _("        -e, --extend             display other/more information\n"));
     fprintf(stderr, _("        -p, --programs           display PID/Program name for sockets\n"));
@@ -1521,6 +1533,9 @@
 	{"verbose", 0, 0, 'v'},
 	{"statistics", 0, 0, 's'},
 	{"numeric", 0, 0, 'n'},
+	{"numeric-hosts", 0, 0, '!'},
+	{"numeric-ports", 0, 0, '@'},
+	{"numeric-users", 0, 0, '#'},
 	{"symbolic", 0, 0, 'N'},
 	{"cache", 0, 0, 'C'},
 	{"fib", 0, 0, 'F'},
@@ -1579,10 +1594,18 @@
 	case 'i':
 	    flag_int++;
 	    break;
-
 	case 'n':
 	    flag_not |= FLAG_NUM;
 	    break;
+	case '!':
+	    flag_not |= FLAG_NUM_HOST;
+	    break;
+	case '@':
+	    flag_not |= FLAG_NUM_PORT;
+	    break;
+	case '#':
+	    flag_not |= FLAG_NUM_USER;
+	    break;
 	case 'N':
 	    flag_not |= FLAG_SYM;
 	    break;
@@ -1645,7 +1668,8 @@
 	    strcpy(afname, DFLT_AF);
 #endif
 	for (;;) {
-	    i = ip_masq_info(flag_not, flag_exp);
+	    i = ip_masq_info(flag_not & FLAG_NUM_HOST,
+			     flag_not & FLAG_NUM_PORT, flag_exp);
 	    if (i || !flag_cnt)
 		break;
 	    sleep(1);
