diff -Nru net-tools-1.52/man/Makefile net-tools-1.52.new/man/Makefile
--- net-tools-1.52/man/Makefile	Wed Apr 21 19:48:54 1999
+++ net-tools-1.52.new/man/Makefile	Fri Jun  4 22:19:49 1999
@@ -23,6 +23,7 @@
 #
 
 INSTALL=install
+mandir=/usr/man
 
 -include ../config.make
 ifeq ($(I18N),1)
@@ -45,9 +46,9 @@
 			cd $$LANG; \
 			for SECTION in 1 5 8; do \
 				if [ "$$LANG" = "en_US" ] ; then \
-					MAN_DIR=${BASEDIR}/usr/man/man$$SECTION; \
+					MAN_DIR=${BASEDIR}$(mandir)/man$$SECTION; \
 				else \
-					MAN_DIR=${BASEDIR}/usr/man/$$LANG/man$$SECTION; \
+					MAN_DIR=${BASEDIR}$(mandir)/$$LANG/man$$SECTION; \
 				fi ; \
 				CMDS=`ls *.$$SECTION`; \
 				[ -z "$$CMDS" ] && continue; \
